dev/plugin-tests.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+## Copyright 2024, gRPC Authors All rights reserved.
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+
+set -euo pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
+error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+fatal() { error "$@"; exit 1; }
+
+source_dir=$(pwd)
```

Comment:
This assumes the current working directory is the root of the package, which is quite brittle.

Most other shell scripts take the location of the shell script and then figure out where the root of the package is relative to that. which is less brittle. Here's an example: https://github.com/grpc/grpc-swift-protobuf/blob/0e6d93b8aaf61270caffc4d6101a48de564b8f11/dev/protos/generate.sh#L18

Reply:
So the reason I didn't do this (which I forgot) was that we tend to invoke scripts by curling and piping them into `bash`. In that case we don't have a source directory to anchor our relative paths to. In that case [our scripts](https://github.com/apple/swift-nio/blob/b14012bb1ae9876e048f60fb6e3986eb061ba558/scripts/check-cxx-interop-compatibility.sh#L24) just assume you're in the right place. I've added some code which attempts to handle both.